uber/site_sections/registration.py

```diff
@@ -885,6 +885,113 @@ def discount(self, session, message='', **params):
 
         return {'message': message}
 
+    @csv_file
+    def generate_promo_codes(self, out, session, **params):
```

Comment:
I'd like to heavily express that when it comes to UX you may throw out all of my work and do whatever makes you happy.